In [1]:
%reload_ext autoreload
%autoreload 2
%reload_ext notexbook
%texify

In [2]:
import numpy as np
import matplotlib.pyplot as plt
import torch
import time
import sys
import pathlib
import pickle
from pathlib import Path
from skimage.io import imread
import seaborn as sns
sns.set_style('white')
sys.setrecursionlimit(10000)
%matplotlib qt5

In [3]:
import cellbgnet
import cellbgnet.utils
from cellbgnet.datasets import DataSimulator
from cellbgnet.utils.hardware import cpu, gpu
from cellbgnet.model import CellBGModel
from cellbgnet.utils.plot_funcs import plot_od, plot_train_record

In [4]:
model_path = Path('/mnt/sda1/SMLAT/training_runs/good_models/model_8865_venus.pkl')

In [5]:
with open(model_path, 'rb') as f:
    chromo_model = pickle.load(f)

In [6]:
chromo_model.evaluation_params['eval_imgs'].shape

(30, 1041, 1302)

In [7]:
plot_train_record(chromo_model)

#### Photon count range, get background average from simulation using model paramteres

In [8]:
mol_photons = (chromo_model.data_generator.simulation_params['min_photon'] + 1) /2 * chromo_model.data_generator.psf_params['photon_scale']

In [9]:
mol_photons

2100.0

#### Plot network outputs for the eval images after tiling and re-tiling of one eval image

In [10]:
from cellbgnet.analyze_eval import recognition, plot_full_img_predictions, assemble_full_img_predictions

In [11]:
eval_img = chromo_model.evaluation_params['eval_imgs'][10][np.newaxis,:]

In [12]:
plt.figure()
plt.imshow(eval_img[0], cmap='gray')
plt.colorbar()
plt.show()

In [13]:
fov_size = [eval_img.shape[2] * 65, eval_img.shape[1] * 65]

In [14]:
fov_size

[84630, 67665]

In [15]:
preds_tmp, n_per_img, plot_data = recognition(model=chromo_model, eval_imgs_all=eval_img,
                                             batch_size=16, use_tqdm=False,
                                             nms=True, candidate_threshold=0.3,
                                             nms_threshold=0.7, 
                                             pixel_nm=chromo_model.data_generator.psf_params['pixel_size_xy'],
                                             plot_num=1,
                                             win_size=128,
                                             padding=True,
                                             start_field_pos=[0, 0],
                                             padded_background=chromo_model.evaluation_params['padded_background'])

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]


In [16]:
plot_full_img_predictions(chromo_model, plot_infs=plot_data, eval_csv=None, plot_num=1, fov_size=fov_size, pixel_size=[65, 65])

#### Testing real image

In [16]:
images_dir = Path('/mnt/sda1/SMLAT/data/real_data/chromosome_dots/pooled_8865/venus/')
img_filenames = sorted(list(images_dir.glob('*.tiff')))

In [60]:
img_path = Path('/mnt/sda1/SMLAT/data/real_data/chromosome_dots/pooled_8865/venus/Pos01_img_000000300.tiff')

In [61]:
real_img = imread(img_path)[np.newaxis, :]


In [62]:
preds_tmp, n_per_img, plot_data = recognition(model=chromo_model, eval_imgs_all=real_img,
                                             batch_size=16, use_tqdm=False,
                                             nms=True, candidate_threshold=0.3,
                                             nms_threshold=0.7, 
                                             pixel_nm=chromo_model.data_generator.psf_params['pixel_size_xy'],
                                             plot_num=1,
                                             win_size=128,
                                             padding=True,
                                             start_field_pos=[0, 0],
                                             padded_background=chromo_model.evaluation_params['padded_background'])

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]


In [20]:
plot_full_img_predictions(chromo_model, plot_infs=plot_data, eval_csv=None, plot_num=1,
                        fov_size=fov_size, pixel_size=[65, 65])

In [63]:
len(preds_tmp)

225

#### Predictions are a list of numbers where each element corresponds to a localizaiton in the following order


    1. counter of the molecule per tile.. not a global counter on the frame, need to fix this
    2. image number used to index into the number of the image in the prediction arrays.
    Eval image is of shape [1, 1041, 1302], so, the image number will always be 1, if you give one image at a time
    3. x position in nm where 0 is top left corner
    4. y position in nm where 0 is top left corner
    5. z position in nm where 0 is from the reference 0 nm in height
    6. photon counts 
    7. probability afer nms
    8. x_sigma in nm 
    9. y_sigma in nm
    10. z_sigma in nm
    11. photon_counts_sigma
    12. x offset
    13. y offset


In [64]:
preds_tmp

[[1.0,
  1.0,
  19589.16015625,
  6858.15625,
  221.94532775878906,
  1282.334716796875,
  0.894707441329956,
  14.139988899230957,
  19.05642318725586,
  39.05344772338867,
  77.52587890625,
  -0.12830111384391785,
  0.010099816136062145],
 [1.0,
  1.0,
  28640.505859375,
  6793.0439453125,
  235.52975463867188,
  1702.05908203125,
  0.9580257534980774,
  13.294410705566406,
  18.742538452148438,
  38.832237243652344,
  221.09327697753906,
  0.12315802276134491,
  0.008365603163838387],
 [1.0,
  1.0,
  37671.44140625,
  6584.79443359375,
  138.81179809570312,
  1399.284423828125,
  1.0151491165161133,
  12.641626358032227,
  15.28670597076416,
  33.92502212524414,
  124.55282592773438,
  0.06063089519739151,
  -0.19546766579151154],
 [1.0,
  1.0,
  42138.265625,
  6786.54248046875,
  214.9449462890625,
  1332.5697021484375,
  0.8346033096313477,
  11.77839469909668,
  19.31284523010254,
  41.28834915161133,
  117.8310546875,
  -0.21896135807037354,
  -0.0916539654135704],
 [1.0,
  1.0

In [57]:
img_infs = assemble_full_img_predictions(chromo_model, plot_data)

In [24]:
img_infs.keys()

dict_keys(['Probs', 'XO', 'YO', 'ZO', 'Int', 'BG', 'XO_sig', 'YO_sig', 'ZO_sig', 'Int_sig', 'Probs_ps', 'XO_ps', 'YO_ps', 'ZO_ps', 'Samples_ps', 'raw_img', 'only_bg'])

In [25]:
for key, value in img_infs.items():
    plt.figure()
    plt.imshow(value)
    plt.colorbar()
    plt.title(key)
    plt.show()

In [27]:
plt.figure()
plt.imshow(img_infs['BG'])
plt.colorbar()
plt.title('Predicted PSF')
plt.show()

In [51]:
plt.figure()
plt.imshow(img_infs['only_bg'][:-200, :-200])
plt.colorbar()
plt.title('Only bg')
plt.show()

##### Collect predictions of all emitters in one image for plotting distributions

In [65]:
x_global = np.array([x[2] for x in preds_tmp])
y_global = np.array([x[3] for x in preds_tmp])
z_global = np.array([x[4] for x in preds_tmp])
ph_global = np.array([x[5] for x in preds_tmp])
prob_dot_nms = np.array([x[6] for x in preds_tmp])
sigma_x = np.array([x[7] for x in preds_tmp])
sigma_y = np.array([x[8] for x in preds_tmp])
sigma_z = np.array([x[9] for x in preds_tmp])
sigma_ph = np.array([x[10] for x in preds_tmp])

#### Only plot z, photon counts and probablitiy so that you can check what is going on as time goes by

In [66]:
fig, ax = plt.subplots(nrows=1, ncols=2)

ax[0].hist(z_global, bins=50)
ax[0].axvspan(z_global.min(), z_global.max(), color='green', alpha=0.1)
ax[0].set_xlabel('z [nm] of the predicted emitters')
ax[0].set_ylabel('counts')

ax[1].hist(ph_global, bins=50)
ax[1].axvspan(ph_global.min(), ph_global.max(), color='green', alpha=0.1)
ax[1].set_xlabel('photon counts of the predicted emitters')
ax[1].set_ylabel('counts')

plt.suptitle(img_path)
plt.show()

### Plotting sigma-xyzi distributions of a single image (sigmas) predicted by the network

In [30]:

fig, ax = plt.subplots(nrows=2, ncols=2)
ax[0, 0].hist(sigma_x, bins=25)
ax[0, 0].axvspan(sigma_x.min(), sigma_x.max(), color='green', alpha=0.1)
ax[0, 0].set_xlabel(r'$\sigma_x$ [nm]')
ax[0, 0].set_ylabel('counts')
ax[0, 0].set_title(f"Mean sigma_x is {np.mean(sigma_x):0.3f}, min:{np.min(sigma_x):0.3f}, max:{np.max(sigma_x):0.3f}")

ax[0, 1].hist(sigma_y, bins=25)
ax[0, 1].axvspan(sigma_y.min(), sigma_y.max(), color='green', alpha=0.1)
ax[0, 1].set_xlabel(r'$\sigma_y$ [nm]')
ax[0, 1].set_ylabel('counts')
ax[0, 1].set_title(f"Mean sigma_y is {np.mean(sigma_y):0.3f}, min:{np.min(sigma_y):0.3f}, max:{np.max(sigma_y):0.3f}")


ax[1, 0].hist(sigma_z, bins=25)
ax[1, 0].axvspan(sigma_z.min(), sigma_z.max(), color='green', alpha=0.1)
ax[1, 0].set_xlabel(r'$\sigma_z$ [nm]')
ax[1, 0].set_ylabel('counts')
ax[1, 0].set_title(f"Mean sigma_z is {np.mean(sigma_z):0.3f}, min:{np.min(sigma_z):0.3f}, max:{np.max(sigma_z):0.3f}")

ax[1, 1].hist(sigma_ph, bins=25)
ax[1, 1].axvspan(sigma_ph.min(), sigma_ph.max(), color='green', alpha=0.1)
ax[1, 1].set_xlabel(r'$\sigma_{ph}$ [nm]')
ax[1, 1].set_ylabel('counts')
ax[1, 1].set_title(f"Mean sigma_ph is {np.mean(sigma_ph):0.3f}, min:{np.min(sigma_ph):0.3f}, max:{np.max(sigma_ph):0.3f}")

plt.show()

#### Plot distributions on predictions in one image xyzi

##### Loop over all images and plot variability in time

In [22]:
number_preds = []
for img_path in img_filenames:
    real_img = imread(img_path)[np.newaxis, :]
    preds_tmp, n_per_img, plot_data = recognition(model=chromo_model, eval_imgs_all=real_img,
                                             batch_size=16, use_tqdm=False,
                                             nms=True, candidate_threshold=0.1,
                                             nms_threshold=0.7, 
                                             pixel_nm=chromo_model.data_generator.psf_params['pixel_size_xy'],
                                             plot_num=1,
                                             win_size=128,
                                             padding=True,
                                             start_field_pos=[0, 0],
                                             padded_background=chromo_model.evaluation_params['padded_background'])
    number_preds.append(len(preds_tmp))
    print(img_path)

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000000.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000001.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000002.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000003.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000038.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000039.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000040.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000041.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000076.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000077.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000078.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000079.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000114.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000115.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000116.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000117.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000152.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000153.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000154.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000155.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000190.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000191.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000192.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000193.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000228.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000229.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000230.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000231.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000266.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000267.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000268.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000269.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000304.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000305.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000306.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000307.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000342.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000343.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000344.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000345.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000380.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000381.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000382.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000383.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000418.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000419.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000420.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000421.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000456.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000457.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000458.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000459.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000494.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000495.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000496.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000497.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000532.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000533.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000534.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000535.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000570.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000571.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000572.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/SMLAT/data/real_data/chromosome_dots/pooled_3024/venus/Pos32_img_000000573.tiff
processing area:99/99, input field_xy:[1258 1301 1004 1043], use_coordconv:True, retain locs in area:[1278, 1301, 1024, 1043]
/mnt/sda2/

In [25]:
plt.figure()
plt.plot(np.arange(0, len(number_preds)), number_preds)
plt.xlabel('Frame number')
plt.ylabel('Number of dots predicted')
plt.title('Chromosome dots 8865 Pos01')
plt.show()